src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,152 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int GROW_FACTOR = 2;
+    private Node<K, V>[] table = new Node[DEFAULT_CAPACITY];
+    private int size = 0;
 
     @Override
     public void put(K key, V value) {
-
+        if (threshold() == size) {
+            resize();
+        }
+        int hash = key == null ? 0 : key.hashCode();
+        Node<K, V> newNode = new Node<>(hash, key, value, null);
+        linkToBucket(table, newNode);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (size == 0) {
+            return null;
+        }
```

Comment:
this is also redundant, your code should handle it without any problems or too big of a time waste, maybe just to calculate hash, and then see right away that the first node is null, thus returning null
```suggestion
```